apps/studio/src/store/modules/TabModule.ts

```diff
@@ -78,6 +86,18 @@ export const TabModule: Module<State, RootState> = {
       context.commit('set', [])
       context.commit('setActive', null)
     },
+    async reopenLastClosedTab(context) {
+      for (let i = context.state.lastClosedTabs.length - 1; i >= 0; i--) {
```

Comment:
Why are we looping here? We only want the very last item in the lastClosedTabs array.

Reply:
When the `lastClosedTabs` is filled, and then we disconnect from current connection to connect to a new connection, we need to search for tabs that belong to the right connection. Using `array.pop` would get the last item, but that item could be from a different `connectionId`. Another solution to this is I was thinking if we should just wipe out the stack so it's empty whenever we connect to a new connection, but maybe it's better to do the first one?